libursa/src/signatures/ed25519.rs

```diff
@@ -1,163 +1,58 @@
-use super::*;
+use super::{SignatureScheme, KeyPairOption, PublicKey, PrivateKey};
 use CryptoError;
+pub use ed25519_dalek::{PUBLIC_KEY_LENGTH as PUBLIC_KEY_SIZE, EXPANDED_SECRET_KEY_LENGTH as PRIVATE_KEY_SIZE, SIGNATURE_LENGTH as SIGNATURE_SIZE};
+use ed25519_dalek::{Signature, Keypair, PublicKey as PK};
+use rand_chacha::ChaChaRng;
+use rand::rngs::OsRng;
+use rand::SeedableRng;
+use sha2::Digest;
 
-pub const PRIVATE_KEY_SIZE: usize = 64;
-pub const PUBLIC_KEY_SIZE: usize = 32;
-pub const SIGNATURE_SIZE: usize = 64;
 pub const ALGORITHM_NAME: &str = "ED25519_SHA2_512";
 
-pub struct Ed25519Sha512(ed25519_sha2_512::Ed25519Sha512Impl);
+pub struct Ed25519Sha512;
 
 impl SignatureScheme for Ed25519Sha512 {
-    fn new() -> Ed25519Sha512 {
-        Ed25519Sha512(ed25519_sha2_512::Ed25519Sha512Impl::new())
+    fn new() -> Self {
+        Self
     }
     fn keypair(&self, option: Option<KeyPairOption>) -> Result<(PublicKey, PrivateKey), CryptoError> {
-        self.0.keypair(option)
+        let kp = match option {
+            Some(o) => match o {
+                KeyPairOption::UseSeed(s) => {
+                    let hash = sha2::Sha256::digest(s.as_slice());
```

Comment:
The caller of `keypair` will lose access to `s`. Should this function be zeroing `s`?

Reply:
ed25519-dalek implements this for you in its Drop trait.